net/http/CMakeLists.txt

```diff
@@ -31,21 +31,21 @@ mark_as_advanced(FASTCGI_INCLUDE_DIR FASTCGI_LIBRARY)
 
 if(FASTCGI_INCLUDE_DIR AND FASTCGI_LIBRARY)
   include_directories(${FASTCGI_INCLUDE_DIR})
+else()
+  set(_nofcgi "ON")
+  set(FASTCGI_LIBRARY "")
 endif()
 
 if(ssl)
   include_directories(${OPENSSL_INCLUDE_DIR})
+  if(OPENSSL_VERSION AND (${OPENSSL_VERSION} VERSION_EQUAL "1.1"))
```

Comment:
Is `OPENSSL_VERSION` guaranteed to end after `1.1`, or can it be `1.1.0e`? If it's the latter, the `VERSION_EQUAL` comparison will fail.

Reply:
cmake macro `VERSION_EQUAL` compares only available parts of the version string.
On my system it is something like "1.1.0a"

Reply:
I tried:
```
# t.cmake
if("1.1.0.1" VERSION_EQUAL "1.1")
message("Hello")
endif()```
```
cmake -P t.cmake
```
prints nothing. If I change `"1.1"` to `"1.1.0.1"` then it matches and prints the message.